src/nupic/math/SparseBinaryMatrix.hpp

```diff
@@ -2002,7 +2002,7 @@ namespace nupic {
 	  << "Invalid index: " << *it
 	  << " - Should be >= 0 and < number of columns:" << nCols();
       
-      for (size_type i = 1; i < (size_type)(end - begin); ++i)
+      for (auto i = 1; i < (end - begin); ++i)
```

Comment:
I tried your suggestion but:

`error C2440: 'initializing': cannot convert from 'nullptr' to 'capnp::_::IndexingIterator<const capnp::List<uint32_t,PRIMITIVE>::Reader,T>'`
`error C2676: binary '&&': 'capnp::_::IndexingIterator<const capnp::List<uint32_t,PRIMITIVE>::Reader,T>' does not define this operator or a conversion to a type acceptable to the predefined operator`

It seems that `InputIterator` is not happy with null references..